viewsb/backends/phywhisperer.py

```diff
@@ -0,0 +1,169 @@
+"""
+PhyWhisperer-USB backend for ViewSB
+Inspired by OpenVizsla backend
+
+
+This file is part of ViewSB
+"""
+
+# pylint: disable=maybe-no-member,access-member-before-definition
+
+import sys
+import errno
+import argparse
+import time
+from datetime import datetime
+
+from ..backend import ViewSBBackend
+from ..packet import USBPacket, USBControlTransfer, USBPacketID
+
+
+try:
+    from phywhisperer import USBEventSink, usb as pw
+
+    class ViewSBEventSink(USBEventSink):
+        """ PhyWhisperer USB event sink that submits packets for decoding. """
+
+
+        def __init__(self, backend, suppress_packet_callback=None):
+
+            # Store a reference to our parent backend, so we can submit USB data via it.
+            self.backend = backend
+
+            # Mark ourselves as having no packet pending.
+            self.pending_packet = None
+
+            # Store the callback we use to determine if we should suppress packets.
+            self.suppress_packet_callback = suppress_packet_callback
+
+
+        def _emit_packet(self, packet):
+            """ Emits a packet to the main decoder thread for analysis. """
+            self.backend.emit_packet(packet)
+
+
+        def _should_be_suppressed(self, packet):
```

Comment:
This is built-in to the main analyzer stack, now. Backends don't explicitly need to handle SOF filtering; so the whole "should be suppressed" mechanism doesn't need to be around. :)

Reply:
removed